apple/Sources/Sargon/Extensions/Swiftified/Profile+Supporting+Types/References/RefProfile+Swiftified.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  File.swift
+//  
+//
+//  Created by Alexander Cyon on 2024-05-01.
+//
+
+import Foundation
+import SargonUniFFI
+
+public protocol SargonReferenceType: SargonModel {
+	associatedtype Inner: SargonModel
+	func take() throws -> Inner
+	static func from(inner: Inner) -> Self
```

Comment:
cannot use `init(inner: Inner)` since `RefBytes` and `RefProfile` is not `final` an those initializers are `convenience`....

Reply:
Can't we make `RefBytes` and `RefProfile` final? is there a reason for these to not be?

Reply:
not possible by UniFFI.... we would need to add more logic to our custom Post Processing step. I lack the Regexp skills :P

Reply:
well, we did this previously, "by hand" for RET types...
And really there is no reason for Sargon objects to be non-final, as those are not meant to be subclassed, though it is interesting what the behaviour is if we do so 😄

Reply:
Nothing should be done by hand. I have created the context where it should happen, the post process script I invented :) I just think we ought to do it with a regexp. I can take a look later.